crates/core/src/client/mod.rs

```diff
@@ -239,15 +247,18 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
             _ => (None, None), // skip all transactions other than Invoke, covers the pending case
         };
         let eth_transaction = transaction.to_eth_transaction(self, block_hash, block_num, None).await?;
-        Ok(eth_transaction)
+        Ok(Some(eth_transaction))
     }
 
     /// Returns the receipt of a transaction by transaction hash.
     async fn transaction_receipt(&self, hash: H256) -> Result<Option<TransactionReceipt>, EthApiError<P::Error>> {
         // TODO: Error when trying to transform 32 bytes hash to FieldElement
         let transaction_hash: Felt252Wrapper = hash.try_into()?;
         let starknet_tx_receipt =
-            self.starknet_provider.get_transaction_receipt::<FieldElement>(transaction_hash.into()).await?;
+            match self.starknet_provider.get_transaction_receipt::<FieldElement>(transaction_hash.into()).await {
+                Err(_) => return Ok(None),
+                Ok(receipt) => receipt,
+            };
 
         let res_receipt = match starknet_tx_receipt {
```

Comment:
Should we make an issue to have a starknet receipt wrapper type in order to implement `to_eth_receipt`?

Reply:
https://github.com/kkrt-labs/kakarot-rpc/issues/311